File: ContractAppAPI/Models/Contract.cs
using System.ComponentModel.DataAnnotations;

namespace ContractAppAPI.Models
{
    public class Contract
    {
        public int Id { get; set; }
        public int ContractNumber { get; set; }
        [MaxLength(100)]
        public string Name { get; set; }
        public DateTime DateOfConclusion { get; set; }
        [MaxLength(500)]
        public string Description { get; set; }
        public decimal Value { get; set; }
        [MaxLength(50)]
        public string Contractor { get; set; }
        [MaxLength(50)]
        public string Signatory { get; set; }
        public Boolean HasPdf { get; set; }
        public int ContractTypeOneId { get; set; }
        public int ContractTypeTwoId { get; set; }
        public ContractTypeOne ContractTypeOne { get; set; }
        public ContractTypeTwo ContractTypeTwo { get; set; }
        public ICollection<ContractPdf> ContractPdfs { get; set; }
        public ICollection<AnnexToTheContract> AnnexToTheContracts { get; set; }
    }
}

File: client/src/app/_models/contract.ts
import { ContractTypeOne } from "./contractTypeOne"
import { ContractTypeTwo } from "./contractTypeTwo"
import { ContractPdf } from "./contractPdf"
import { AnnexToTheContract } from "./annexToTheContract"


export interface Contract {
    id: number
    contractNumber: number
    name: string
    typeNameOne: string
    typeNameTwo: string
    dateOfConclusion: string
    description: string
    value: number
    contractor: string
    signatory: string
    hasPdf: boolean

    contractTypeOne: ContractTypeOne
    contractTypeTwo: ContractTypeTwo
    contractPdfs: ContractPdf[]
    annexToTheContract: AnnexToTheContract[]
  }